### ISSUE TITLE
Marlin-bugfix-2.0.x-bug

### ISSUE THREAD

**Issue by hn0550** [MarlinFirmware/Marlin#8368]
### Compile Error
1. user  `MOTHERBOARD BOARD_RAMPS_FD_V1`
```
err:  C:\Users\Administrator\AppData\Local\Arduino15\packages\arduino\tools\arm-none-eabi-gcc\4.8.3-2014q1/bin/arm-none-eabi-gcc.exe: The filename or extension is too long.
```
2. user  `MOTHERBOARD BOARD_RAMPS_FD_V2`
```
 fatal error: pins_RAMPS_FD.h: No such file or directory
```

**Comment by RowanMeara** [comment]
I think this is an Arduino IDE windows issue.  WIndows has a short path length limit.  There should be a fair bit of discussion on this on the arduino website.

**Comment by hn0550** [comment]
But the previous  version( Marlin-bugfix-2.0.x  2017-09-11) not have this problem。
my syetem: windows 10 
IDE:1.8.4

**Comment by hn0550** [comment]
This confirmation is a mistake :
user MOTHERBOARD BOARD_RAMPS_FD_V2
 fatal error: pins_RAMPS_FD.h: No such file or directory

**Comment by fiveangle** [comment]
You're using old file structure somewhere that has not picked up the file name change in #7852

Also, once you fix that, you'll need to fix the issue that @RowanMeara mentioned. I suggest to file an issue with Arduino project for the command length limit issue (search their github: there are a few issues open against it).

**Comment by hn0550** [comment]
old file structure ? Refers to the?
windows? IDE ? Marlin2.0?  @fiveangle

**Comment by Tannoo** [comment]
In `marlin\src\pins\pins_RAMPS_FD_V2.h`, line 32:

```
#include "pins_RAMPS_FD.h"
```

needs to be changed to:

```
#include "pins_RAMPS_FD_V1.h"
```

That only fixes the pins file issue, but the rest of the build is incomplete.

**Comment by Tannoo** [comment]
>user MOTHERBOARD BOARD_RAMPS_FD_V1
err: C:\Users\Administrator\AppData\Local\Arduino15\packages\arduino\tools\arm-none-eabi-gcc\4.8.3-2014q1/bin/arm-none-eabi-gcc.exe: The filename or extension is too long.

I didn't have this issue with PlatformIO.
It's not a Windoze issue. I run Win10.

**Comment by fiveangle** [comment]
> In marlin\src\pins\pins_RAMPS_FD_V2.h, line 32:
> 
> #include "pins_RAMPS_FD.h"
> 
> needs to be changed…

You're more tolerant of digging into the code on a whim :) .  It's so much more likely to be a wild goose chase, so I usually reserve that for when they at least upload their config ;) .  But yeah, appears the V1 file name change was referred to in the V2 pins so didn't get picked up when the user submitted #7852.  I always do a global search across the entire project when I do a change like that, just in case there are others lurking.

> It's not a Windoze issue. I run Win10.

???  What is it from then if not the 32k command length limit of Windows causing the Arduino IDE build to fail ? Everyone has been hitting this and I'm pretty sure that was the error.

https://github.com/arduino/Arduino/issues/6751

**Comment by Tannoo** [comment]
I was testing by compiling with those motherboards.

I was using my code used for my printer. I found that V1 had fussed about filament runout sensor. So I diabled that. Then it fused about a variable definition (something to do with max eeprom size for ubl). No path errors.

I set the runour sensor on again and then tried v2... I got the pins file error. Hunted that down. It then fussed about the filament runout sensor.. disabled it. Then fussed about the same definition that v1 did.

I stopped there thinking along the same lines as you about unfinished DUE support.

**Comment by Tannoo** [comment]
I believe the path issue is with Arduino IDE.

In the post you linked.. "...maximum command length of 32767 characters on Windows."

It didn't look like his path error was THAT long.

**Comment by Bob-the-Kuhn** [comment]
The path length issue is being fixed in Arduino 1.9.  The Beta version has it now.

**Comment by fiveangle** [comment]
I linked to the beta version above * shrug *
